internal/agent/sender/sender.go

```diff
@@ -1,78 +1,67 @@
 package sender
 
 import (
-	"bytes"
-	"compress/gzip"
-	"encoding/json"
+	"context"
 	"fmt"
-	"github.com/go-resty/resty/v2"
 	"github.com/maybecoding/go-metrics.git/internal/agent/app"
 	"github.com/maybecoding/go-metrics.git/pkg/logger"
-	"strings"
+	"sync"
 	"time"
 )
 
 type Sender struct {
 	endpoint       string
 	retryIntervals []time.Duration
+	hashKey        string
+	ctx            context.Context
+	numWorkers     int
 }
 
-func (j *Sender) Send(metrics []*app.Metrics) {
-	// Получаем json для отправки
-	payload, err := json.Marshal(metrics)
-	if err != nil {
-		logger.Error().Err(err).Msg("error due marshal all metrics before send")
-		return
+func (j *Sender) SendWorker(inpM chan *app.Metrics, id int) {
+	logger.Debug().Int("number", id).Msg("Started worker")
+	defer func() {
+		logger.Debug().Int("number", id).Msg("Stopped worker")
+	}()
+	for {
+		select {
+		case <-j.ctx.Done():
+			return
+		case m, ok := <-inpM:
+			if !ok {
+				return
+			}
+			j.sendMetric(m)
+		}
 	}
-	logger.Debug().Str("json", string(payload)).Msg("trying to send json")
-
-	// Создаем сжатый поток
-	buf := bytes.NewBuffer(nil)
-	zw := gzip.NewWriter(buf)
+}
 
-	// И записываем в него данные
-	_, err = zw.Write(payload)
-	if err != nil {
-		logger.Error().Err(err).Msg("can't write into gzip writer")
-		return
-	}
-	err = zw.Close()
-	if err != nil {
-		logger.Error().Err(err).Msg("can't close gzip writer")
-		return
-	}
+func (j *Sender) SendStart(inpM chan *app.Metrics) {
```

Comment:
Start предполагает исключительно старт и выход из метода - часто прокидывают контекст с таймаутом чтобы сделать ограничение на страрт - не успели за 3 секнуды - не смогли стартануть - ошибка.

А тут не только старт но и сама работа горутин, причем с блокировкой пока они работают. Обычно такое Run называют

```suggestion
func (j *Sender) Run(inpM chan *app.Metrics) {
```